ui/app/components/mount-backend-form.ts

```diff
@@ -58,18 +67,20 @@ export default class MountBackendForm extends Component<Args> {
     super.willDestroy();
   }
 
-  checkPathChange(type: string) {
-    if (!type) return;
+  checkPathChange(backendType: string) {
+    if (!backendType) return;
     const mount = this.args.mountModel;
     const currentPath = mount.path;
-    const mountTypes =
-      this.args.mountType === 'secret'
-        ? allEngines().map((engine) => engine.type)
-        : methods().map((auth) => auth.type);
+    // while mountType is always set to 'secret' or 'auth', there's situations where we pass in an empty string (such as on setMountType) and in this case we want to default to returning auth methods.
+    const mountsByType = filterEnginesByMountType(
+      this.args.mountType ? this.args.mountType : 'auth',
+      true
+    ).map((engine) => engine.type);
```

Comment:
We can use` ??`, which returns the right-hand value **only if** the left-hand value is null or undefined.
Also, I've made `filterEnginesByMountType` use named params, which makes it easier for a user to understand what we're passing in. In this example, it clarifies what "true" indicates.
```suggestion
    const mountsByType = filterEnginesByMountType({
      mountType = this.args.mountType ??  'auth',
      includeEnterprise = true
    }).map((engine) => engine.type);
```